pkg/operator/csi/csicontrollerset/csi_controller_set.go

```diff
@@ -84,7 +84,7 @@ func (c *CSIControllerSet) WithLogLevelController() *CSIControllerSet {
 // WithManagementStateController returns a *ControllerSet with a management state controller initialized.
 func (c *CSIControllerSet) WithManagementStateController(operandName string, supportsOperandRemoval bool) *CSIControllerSet {
 	c.managementStateController = management.NewOperatorManagementStateController(operandName, c.operatorClient, c.eventRecorder)
-	if supportsOperandRemoval {
+	if !supportsOperandRemoval {
```

Comment:
fun

Reply:
> fun. do you know ahead of time if this breaks someone and/or how it does so?

I can see this is only in your package, but you could consider adding another param to break someone doing this upgrade in a way that they notice.  `reallySupportsOperandRemoval`